module-system/sov-modules-core/src/common/gas.rs

```diff
@@ -34,12 +43,28 @@ impl<const N: usize> GasUnit for TupleGasUnit<N> {
         unit
     }
 
+    fn into_dimensions(&self) -> Vec<u64> {
+        self.to_vec()
+    }
+
     fn value(&self, price: &Self) -> u64 {
         self.iter()
             .zip(price.iter().copied())
             .map(|(a, b)| a.saturating_mul(b))
             .fold(0, |a, b| a.saturating_add(b))
     }
+
+    fn combine(&mut self, rhs: &Self) {
+        for i in 0..N {
+            self[i] = self[i].saturating_add(rhs[i]);
+        }
```

Comment:
nit: This might insert unneeded bounds checks, and I think it's slight more idiomatic to use an iterator anyway. Is there any problem with doing that?

Reply:
It's mostly a case of style since we use a const `N`, so the compiler will end up optimizing both versions to the maximum. But indeed, we use iterators elsewhere on the same source, so it's best to use them here to keep consistency. Fixed!